src/test/integration/utils/wait.js

```diff
@@ -0,0 +1,165 @@
+const { info } = require("./shared");
+
+/**
+ * Waits for `predicate(state)` to be true. `state` is the redux app state.
+ *
+ * @memberof mochitest/waits
+ * @param {Object} dbg
+ * @param {Function} predicate
+ * @return {Promise}
+ * @static
+ */
+async function waitForState(dbg, predicate) {
+  return new Promise(resolve => {
+    const unsubscribe = dbg.store.subscribe(() => {
+      if (predicate(dbg.store.getState())) {
+        unsubscribe();
+        resolve();
+      }
+    });
+  });
+}
+
+async function waitForPaused(dbg) {
+  // We want to make sure that we get both a real paused event and
+  // that the state is fully populated. The client may do some more
+  // work (call other client methods) before populating the state.
+  await waitForThreadEvents(dbg, "paused"),
+  await waitForState(dbg, state => {
+    const pause = dbg.selectors.getPause(state);
+    // Make sure we have the paused state.
+    if (!pause) {
+      return false;
+    }
+    // Make sure the source text is completely loaded for the
+    // source we are paused in.
+    const sourceId = pause.getIn(["frame", "location", "sourceId"]);
+    const sourceText = dbg.selectors.getSourceText(dbg.getState(), sourceId);
```

Comment:
probably not... but i don't want to mess with it because we are only using this in one test now...